### PR TITLE
service catalog documentation updates

### PR DESCRIPTION
**Pull request by jpeeler**
This updates some of the outdated information and includes a few additional pieces of information.

@adellape @pmorie see what you think. Additional renaming consistency may be necessary due to all the renaming changes of late.

### CODE REVIEW

File: architecture/service_catalog/index.adoc [hunk 1]
```diff
@@ -87,6 +87,17 @@ This infrastructure allows a loose coupling between applications running in
 those services to focus on its own business logic while leaving the management
 of these services to the provider.
 
+=== Deleting resources
```
Comment (ahardin-rh) [inline architecture/service_catalog/index.adoc]: Resources

File: architecture/service_catalog/index.adoc [hunk 2]
```diff
@@ -102,16 +113,22 @@ broker available within their cluster.
 +
 A `Broker` resource specifies connection details for a service broker and the
 set of services (and variations of those services) to {product-title} that
-should then be made available to users.
+should then be made available to users. Of special note is the _authInfo_
```
Comment (ahardin-rh) [inline architecture/service_catalog/index.adoc]: `authInfo` section, which contains

File: architecture/service_catalog/index.adoc [hunk 3]
```diff
@@ -87,6 +87,17 @@ This infrastructure allows a loose coupling between applications running in
 those services to focus on its own business logic while leaving the management
 of these services to the provider.
```
Comment (ahardin-rh) [inline architecture/service_catalog/index.adoc]: We need a section ID before the heading:
[[service-catalog-deleting-resources]]

File: architecture/service_catalog/index.adoc [hunk 4]
```diff
@@ -87,6 +87,17 @@ This infrastructure allows a loose coupling between applications running in
...
 
+=== Deleting resources
+
+When a user is done with a service, the service instance can be deleted. In
```
Comment (ahardin-rh) [inline architecture/service_catalog/index.adoc]: Why would users want to delete resources? To free up memory resources? If they're not deleted, what can be expected?

File: architecture/service_catalog/index.adoc [hunk 5]
```diff
@@ -185,15 +204,44 @@ in which pods.
...
   secretName: mySecret
-  <pod_selector_labels>
 ----
 
+Parameters::
+A _parameter_ is a special field available to pass additional data to the broker 
+when using either bindings or instances. The only formatting requirement is for
+the parameters to be valid YAML (or JSON). In the above example, a security
+level parameter is passed to the broker in the binding request. For parameters
+that need more security, the parameter(s) should be placed in a secret and then
```
Comment (ahardin-rh) [inline architecture/service_catalog/index.adoc]: For parameters that need more security, place them in a secret and reference them using `parametersFrom`.

File: architecture/service_catalog/index.adoc [hunk 6]
```diff
@@ -185,15 +204,44 @@ in which pods.
...
+referenced using `parametersFrom`.
++
+.Example Binding Resource referencing secret
```
Comment (ahardin-rh) [inline architecture/service_catalog/index.adoc]: Referencing a Secret

**Comment by ahardin-rh** [comment]
@jpeeler Just a few comments from me. Changes will have to be squashed into 1 commit for publishing. Thanks!

**Comment by jpeeler** [comment]
@ahardin-rh I've revised into one commit along with your requested changes. See if you think all is well now.

**Comment by ahardin-rh** [comment]
@jpeeler LGTM. Thanks!

**Comment by ahardin-rh** [comment]
No revision history needed

**Pull request merged**